addons/mail_plugin/models/ir_http.py

```diff
@@ -24,3 +26,8 @@ def _auth_method_outlook(cls):
 
         # take the identity of the API key user
         request.uid = user_id
+
+        lang = request.env['res.users'].browse(user_id).sudo().lang
+        if lang and lang != request.env.context.get('lang', 'en_US'):
+            # switch to the user lang
+            request.env.context = frozendict({**request.env.context, 'lang': lang})
```

Comment:
going to conflict in 15.3 because of httpocalypse, to update the context in 15.3 you can do instead: `request.update_context(lang=lang)`

you will also be able to replace `request.env["res.users"].browse(...)` by `request.env.user`